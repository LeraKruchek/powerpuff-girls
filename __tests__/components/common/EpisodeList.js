import React from 'react';
import { shallow, configure } from 'enzyme';
import { mountToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import EpisodeList from '../../../src/components/common/EpisodesList';

configure({ adapter: new Adapter() });

// unit tests for the App component
describe('EpisodeList component', () => {
  describe('render()', () => {
    it('should render the component', () => {
      const props = { episodes: [] };
      const wrapper = shallow(<EpisodeList {...props} />);
      expect(wrapper.length).toEqual(1);
      expect(mountToJson(wrapper)).toMatchSnapshot();
    });

    it('should load more on button click', () => {
      const mockLoadMore = jest.fn();
      const props = { episodes: [] };
      const wrapper = shallow(<EpisodeList {...props} />);
      wrapper.instance().loadMore = mockLoadMore;
      wrapper.instance().forceUpdate();
      wrapper.update();

      wrapper.find('button').simulate('click');

      expect(mockLoadMore).toHaveBeenCalled();
      expect(wrapper.length).toEqual(1);
    });
  });
});
