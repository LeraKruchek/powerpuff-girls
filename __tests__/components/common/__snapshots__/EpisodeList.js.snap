// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EpisodeList component render() should render the component 1`] = `
<div
  className="episodes-container"
>
  <table
    className="table table-sm"
  >
    <thead>
      <tr>
        <th
          scope="col"
        >
          Season
        </th>
        <th
          scope="col"
        >
          Episode â„–
        </th>
        <th
          scope="col"
        >
          Episode name
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Load more
  </button>
</div>
`;
